test/unit/services/test_catalogue_category_property.py

```diff
@@ -169,11 +184,55 @@ def test_create_mandatory_property_without_default_value(
     item_repository_mock.insert_property_to_all_in.assert_not_called()
 
 
+def test_create_non_existent_unit_id(
+    test_helpers,
+    catalogue_category_repository_mock,
+    catalogue_item_repository_mock,
+    item_repository_mock,
+    unit_repository_mock,
+    catalogue_category_property_service,
+):
+    """
+    Test creating a property at the catalogue category with a non existent unit id
+    """
+    catalogue_category_id = str(ObjectId())
+    unit = UnitOut(id=str(ObjectId()), **UNIT_A)
+    catalogue_item_property = CatalogueItemPropertyPostRequestSchema(
+        name="Property A", type="number", unit_id=unit.id, mandatory=False
```

Comment:
This can just be `unit_id` as that is the only thing required.
```suggestion
    unit_id = str(ObjectId())
    catalogue_item_property = CatalogueItemPropertyPostRequestSchema(
        name="Property A", type="number", unit_id=unit_id, mandatory=False
```